src/Nethermind/Nethermind.Core/Crypto/KeccakCache.cs

```diff
@@ -130,7 +130,7 @@ private static uint FastHash(ReadOnlySpan<byte> input)
 
         ref var b = ref MemoryMarshal.GetReference(input);
 
-        // Start with first
+        // Start with instance random, length and first as seed
```

Comment:
@benaadams The length is not needed I believe as it provides not that much entropy. Also, the lenght later is encoded as a part of `HashAndLength` so can be compared directly in one comparison.

Reply:
instance random will be same, [Benford's law](https://en.wikipedia.org/wiki/Benford%27s_law) says most of the leading bytes on big endian will skew the same (hence my other comment on high entropy at the end, rather than start; and more so if leading zeros aren't trucated), so in-order to not frequently start with same seed I'm incorporating length.

Reply:
i.e. including .Length gives more value in running the first Crc